src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -1285,7 +1285,8 @@ private ImmutableArray<ParameterSymbol> ComputeParameters(Binder binder, BasePro
                 }
                 else if ((object)_setMethod != null && param.Name == ParameterSymbol.ValueParameterName)
                 {
-                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, _location, param.Name);
+                    // note: there can only ever be one location for param, as it cannot be used in partial for a property
+                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, param.Locations[0], param.Name);
```

Comment:
RE: param.Locations?[0] valid syntax?

It is valid syntax, but semantically incorrect. Locations property returns a struct. Besides, even if Locations was returning a regular array, that would check for null array rather than for an empty array. We wan't to make sure an empty array is handled.